src/Microsoft.SqlTools.ServiceLayer/EditData/EditTableMetadata.cs

```diff
@@ -55,6 +59,39 @@ public EditTableMetadata()
 
         #endregion
 
+        // Filters out metadata that is not present in the result set, and matches metadata ordering to resultset
+        public static EditColumnMetadata[] FilterColumnMetadata(EditColumnMetadata[] metaColumns, DbColumnWrapper[] resultColumns)
+        {
+            if(metaColumns.Length == 0)
+            {
+                return metaColumns;
+            }
+
+            bool escapeColName = metaColumns[0].EscapedName.Contains("[");
```

Comment:
As per discussion, let's try to keep name escaping and whatnot inside the ToSqlScript/FromSqlScript classes